chall04/ptuukkan.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/python3
+# Version 3.6.8
+
+import sys
+
+def validate_input(lines):
+	if len(lines) == 0:
+		return 0
+	dim = lines.pop(0)
+	if not dim.isnumeric():
+		return 0
+	dim = int(dim)
+	if len(lines) != dim:
+		return 0
+	for line in lines:
+		if len(line) != dim:
+			return 0
+		for c in line:
+			if not c in " .#":
+				return 0
+	return dim
+
+def convert_lines(lines, dim):
+	"Changes list lines from horizontal to vertical"
+	i = 0
+	converted_lines = []
+	while i < dim:
+		converted_lines.append("".join(line[i] for line in lines))
```

Comment:
converted_lines = ["".join(line[i] for line in lines) for i in range(dim)]
the whole function in one readable line — that's the Python power ;)